guides/source/upgrading_ruby_on_rails.md

```diff
@@ -169,6 +169,86 @@ more meaningful preview images. Previously the first frame of the video would be
 and that caused problems if the video faded in from black. This change requires
 FFmpeg v3.4+.
 
+### ActiveStorage variant processor changed to :vips
+
+Image transformation will now use libvips instead of ImageMagick. This will reduce
+the time taken to generate variants as well as CPU and memory usage, improving response
+times in apps that rely on active storage to serve their images. 
+
+The `:mini_magick` option is not being deprecated, so it is fine to keep using it.
+
+Migrating to libvips requires changing existing image transformation code to the
+`image_processing` macros, and replacing ImageMagick's options with libvips' options.
+
+#### Replace resize with resize_to_limit
+```ruby
+variant(resize: "100x")
+variant(resize_to_limit: [100, nil])
+```
```

Comment:
```suggestion
```diff
- variant(resize: "100x")
+ variant(resize_to_limit: [100, nil])
```
```

did you mean this?

Reply:
Ah, much better. I'll fix it.

Reply:
As to how to preview it... I'm not really sure.

Reply:
https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#action-cable-javascript-api-changes

This looks correct:

![image](https://user-images.githubusercontent.com/509837/125356022-4ecba580-e32b-11eb-8864-0f225106e763.png)